apps/explorer/lib/explorer/third_party_integrations/auth0/legacy.ex

```diff
@@ -0,0 +1,235 @@
+           provider: provider,
+           user_id: secondary_identity_id
+         },
+         {:ok, %OAuth2.Response{status_code: 201}} <-
+           Client.post(
+             client,
+             "#{Internal.users_path()}/#{URI.encode(primary_user_id)}/identities",
+             body,
+             Internal.json_content_type()
+           ) do
+      :ok
+    else
+      error -> Internal.handle_common_errors(error, "Failed to link accounts")
+    end
+  end
+
+  defp update_user_email(user_id, email) do
+    Internal.update_user(user_id, %{"user_metadata" => %{"email" => email}}, "Failed to update user email")
+  end
+
+  defp maybe_link_email(%{"email" => email, "user_id" => "email|" <> identity_id = user_id} = user) do
+    case Internal.find_users(
+           ~s(email:"#{URI.encode(email)}" AND NOT user_id:"#{URI.encode(user_id)}"),
+           "Failed to find legacy users by email"
+         ) do
+      {:ok, []} ->
+        {:ok, user}
+
+      {:ok, [%{"user_id" => primary_user_id} = user]} ->
+        link_users(primary_user_id, identity_id, "email")
+        maybe_verify_email(user)
+        {:ok, user}
+
+      {:ok, users} ->
+        merge_email_users(users, identity_id, "email")
+
+      error ->
+        error
+    end
+  end
+
+  defp maybe_link_email(user) do
+    {:ok, user}
+  end
+
+  defp maybe_verify_email(%{"email_verified" => false, "user_id" => user_id}) do
+    with {:ok, _} <-
+           Internal.update_user(user_id, %{"email_verified" => true}, "Failed to patch email_verified to true") do
+      :ok
+    end
+  end
+
+  defp maybe_verify_email(_), do: :ok
+
+  defp merge_email_users([primary_user | _] = users, identity_id_to_link, provider_for_linking) do
+    identity_map =
+      users
+      |> Enum.map(& &1["user_id"])
+      |> Identity.find_identities()
+      |> Map.new(&{&1.uid, &1})
+
+    users_map = users |> Enum.map(&{&1["user_id"], &1}) |> Map.new()
+
+    case users |> Enum.map(&identity_map[&1["user_id"]]) |> Enum.reject(&is_nil(&1)) |> Account.merge() do
+      {{:ok, 0}, nil} ->
+        link_users(primary_user["user_id"], identity_id_to_link, provider_for_linking)
+        maybe_verify_email(primary_user)
+        {:ok, primary_user}
+
+      {{:ok, _}, primary_identity} ->
+        link_users(primary_identity.uid, identity_id_to_link, provider_for_linking)
+        maybe_verify_email(users_map[primary_identity.uid])
+        {:ok, users_map[primary_identity.uid]}
+
+      error ->
+        Logger.error(["Error while merging users with the same email: ", inspect(error)])
+        :error
+    end
+  end
+
+  defp merge_web3_users([primary_user | _] = users) do
+    identity_map =
+      users
+      |> Enum.map(& &1["user_id"])
+      |> Identity.find_identities()
+      |> Map.new(&{&1.uid, &1})
+
+    users_map = users |> Enum.map(&{&1["user_id"], &1}) |> Map.new()
+
+    case users |> Enum.map(&identity_map[&1["user_id"]]) |> Enum.reject(&is_nil(&1)) |> Account.merge() do
```

Comment:
I personally would pipe here into case as well, but you can surely discard my comment, as the code is already perfect. 

```suggestion
  users
  |> Enum.map(&identity_map[&1["user_id"]])
  |> Enum.reject(&is_nil(&1))
  |> Account.merge()
  |> case do
```